### DISCUSSION TITLE
Ab und zu wird der Wert falsch interpretiert

### DISCUSSION THREAD

**Discussion by ppmk2** [jomjol/AI-on-the-edge-device#2093] [Q&A]
Hallo 
Ich habe das ganzen nun 4 Tage am Wasserzähler laufen und ab und zu werden die Digi nicht richtig erkannt.
satt 45,xxxx m³ werden 44,xxxx ausgegeben. Wenn ich dann mal auf recognition drücke wir es immer richtig angezeigt.
Hier mal die letzten Einträge der influxdb
time                                ack  from                                q value
1677493636095000000 true system.adapter.mqtt.0     0 44.9152
1677493926997000000 true system.adapter.mqtt.0     0 44.9152
1677494218030000000 true system.adapter.mqtt.0     0 45.9152
1677495384601000000 true system.adapter.mqtt.0     0 45.9152
1677495673166000000 true system.adapter.mqtt.0     0 44.9152
1677495963991000000 true system.adapter.mqtt.0     0 45.9152
1677496254967000000 true system.adapter.mqtt.0     0 44.9152

![20230227_1152](https://user-images.githubusercontent.com/126456937/221545005-cd363e8d-1f10-418d-924d-18288a561da2.jpg)

Kann man da noch was einstellen, das es besser funktioniert?

Andere Lichteinflüsse sollten nicht das Problem sein.

![WhatsApp Image 2023-02-27 at 12 30 31](https://user-images.githubusercontent.com/126456937/221553147-378fa38b-852f-4951-b9cd-315211734c4e.jpeg)

Ich habe jetzt auch nicht die aktuellste Version installiert, ich hatte mich schon einmal daran versucht und aufgegeben.  
Die installierte Version ist v11.2.0 - 2022-08-28.
An einem Update hatte ich mich auch schon mal versucht, hat aber leider nicht funktioniert.
Gibt es da eine Anleitung für dummies?

Grüße

**Comment by caco3** [comment]
v11.2.0  ist schon sehr alt, da hat sich sehr viel getan, ins. an der stabilität und error handling. mach doch bitte ein update auf v15.0!

a der sprung so gross ist, solltest du am besten die config verwerfen und neu starten.

Das aufsetzen ist mittlerweile viel einfache geworden, siehe https://jomjol.github.io/AI-on-the-edge-device-docs/Installation/#2-firmware

**Comment by ppmk2** [comment]
Vielen Dank caco3 

Also auf Version 13 hab ich das wohl geschafft. 
![Screenshot 2023-02-27 171050](https://user-images.githubusercontent.com/126456937/221617457-24c7e2ce-9dae-47fc-808b-c6fa1cf069ba.jpg)
Wenn ich jetzt aber versuche die 15 zu laden, kommt das:
![error](https://user-images.githubusercontent.com/126456937/221617835-bd377798-2c28-48b3-b9fa-ef48d48877d1.jpg)

Was macht man da?

**Reply by caco3** [reply]
wieso hast Du v14.0.3 ausgelassen?
Ich weiss nicht, ob ein update von 13 auf 15 geht, das haben wir nie gestestet!

**Comment by ppmk2** [comment]
Ach so, das war mir nicht bewusst, das ich erst 14 und 15 nehmen muss.  
Das tut mir echt leid das ich mich so dusselig anstelle, aber wo finde ich denn jetzt die Version 14?
Auf der Seite [https://github.com/jomjol/AI-on-the-edge-device/releases](url) finde ich sie nicht.

**Reply by caco3** [reply]
Kein Problem!

Du must etwas nach unten scrollen.

Oder gleich hier drauf: https://github.com/jomjol/AI-on-the-edge-device/releases/tag/v14.0.3 :)

**Comment by ppmk2** [comment]
Ich mach wohl trotzdem was falsch.
Ich habe jetzt die Version 14 heruntergeladen und entpackt, über durchsuchen die firmware.bin aus dem Ordner  AI-on-the-edge-device__manual-setup__v14.0.3 ausgewählt und bekomme die Meldung:
![error3](https://user-images.githubusercontent.com/126456937/221695473-23cfc050-8dd5-4a7a-bc79-d5b973ea2877.jpg)
Mach ich das nicht richtig?

Vielen Dank für deine Geduld.

**Reply by caco3** [reply]
spiel das ganze v14 update zip ein.
Das Web UI sollte immer zusammen aktualisiert werden.

Falls da aber nicht mehr geht, müsstest du die v15 firmware via UART (z.B. Web Installer) einspielen

**Comment by ppmk2** [comment]
Muss ich wohl neu via UART machen.
![error4](https://user-images.githubusercontent.com/126456937/221924967-08c1f3fb-0f38-48a3-bf4d-b441fef261a5.jpg)

Ach nach dem Neustart immer das gleiche.

**Comment by ppmk2** [comment]
Um das richtig zu verstehen. 
Mit UART ist wie ich die 13ner Version drauf bekommen habe?
Und jetzt mach ich das noch einmal so nur das ich die  bootloader.bin und partitions.bin aus dem Ordner v11.2.0 nehme und die firmware.bin aus dem 15.0.1?

**Reply by caco3** [reply]
Bitte mach die Installation mit v15 gemäss Anleitung in https://jomjol.github.io/AI-on-the-edge-device-docs/Installation/#2-firmware
Dann must du selber gar nichts mit irgendwelchen Dateien machen!

**Reply by ppmk2** [reply]
Ich habe es jetzt auf 15.0.2 geschafft. 
Jetzt läuft das alles schon viel besser. Mal sehen wie es in ein paar Tagen aussieht.
Vielen Dank :-)

**Comment by ghost** [comment]
> Hallo Ich habe das ganzen nun 4 Tage am Wasserzähler laufen und ab und zu werden die Digi nicht richtig erkannt. satt 45,xxxx m³ werden 44,xxxx ausgegeben. Wenn ich dann mal auf recognition drücke wir es immer richtig angezeigt. Hier mal die letzten Einträge der influxdb time ack from q value 1677493636095000000 true system.adapter.mqtt.0 0 44.9152 1677493926997000000 true system.adapter.mqtt.0 0 44.9152 1677494218030000000 true system.adapter.mqtt.0 0 45.9152 1677495384601000000 true system.adapter.mqtt.0 0 45.9152 1677495673166000000 true system.adapter.mqtt.0 0 44.9152 1677495963991000000 true system.adapter.mqtt.0 0 45.9152 1677496254967000000 true system.adapter.mqtt.0 0 44.9152

Zu deiner ursprünglichen Frage: 
Was passiert, ist dass das Netz mal 5.0 mal 4.7 erkennt. Es gibt im Dataset leider aus früherer Zeit viele versetzte Bilder, deshalb 5.0, da früher alles nur ganze Zahlen waren.

Die ROIs (nur digital) könnten besser justiert werden, da das Model nicht immer den Rand erkennt. Hier können leichte Lichtschankungen schon unterschiedliche Ergebnisse bringen.
https://jomjol.github.io/AI-on-the-edge-device-docs/ROI-Configuration/

Das Bild zeigt es eigentlich ganz gut:

![](https://user-images.githubusercontent.com/412645/199028748-c48ef5bb-a8d4-4c77-9faf-763e6cf77351.png)


Und es gibt ganz viel Anleitung unter https://jomjol.github.io/AI-on-the-edge-device-docs/

**Reply by ppmk2** [reply]
Oh so exakt muss das sein? 
Ich dachte erst der Fehler kommt davon, wenn die Zahl sich aus dem kleineren roten Kästchen bewegt, das sie nicht mehr richtig gelesen wird.
Dann werde ich das noch einmal richtig positionieren.

Aber nach dem Firmware update ist es auch schon viel besser geworden.
Danke :-)

**Reply by ghost** [reply]
Man muss sich vorstellen, dass auf dem kleinen Gerät ja nur eben dieser kleine Ausschnitt in die AI gegeben wird. Wer das selbst mal versucht hat zu bestimmen, merkt wie schwer man sich bei den Zahlen tut.

**Comment by FFellheim** [comment]
> Hallo Ich habe das ganzen nun 4 Tage am Wasserzähler laufen und ab und zu werden die Digi nicht richtig erkannt. satt 45,xxxx m³ werden 44,xxxx ausgegeben. ...<

Hallo ppmk2,
dein Ursprungs-Post ist zwar schon vom 27.02.2023, dennoch ein Hinweis:
Die richtige Interpretation des Zählerstandes lautet 44,9152 und NICHT 45,9152. Die 45 m³ sind noch nicht voll erreicht, weil dein Zähler tatsächlich erst 44,9xxx anzeigt.
Die AI interpretiert also richtig😄, sofern sie 44,9 anzeigt!
Viele Grüße
